### PR TITLE
feat(cloudwatch): add `TableWidget`

### LINKED ISSUE #28975: cloudwatch: support the new data table widget
### Describe the feature

Add support for the newly released data table widget https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/add_remove_table_dashboard.html

### Use Case

Leverage data table widget in CloudWatch Dashboards

### Proposed Solution

_No response_

### Other Information

The full schema is not yet posted in documentation at https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/CloudWatch-Dashboard-Body-Structure.html

I’ve asked CW team to update this page so work in CDK can be based off of that

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.125.0

### Environment details (OS name and version, etc.)

macOS

### PR DESCRIPTION
**Pull request by humanzz**
### Issue # (if applicable)

closes #28975.

### Reason for this change

add support for table widget https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/add_remove_table_dashboard.html

### Description of changes

add a new `TableWidget` and its supporting property classes/interfaces

### Description of how you validated changes

added both unit/integ tests

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by humanzz** [comment]
In lack of an updated JSON schema - which I've asked the cloudwatch team to provide - in https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/CloudWatch-Dashboard-Body-Structure.html, I went ahead with this implementation based on
- documentation/examples given in https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/add_remove_table_dashboard.html
- manual experimentation via console

**Review by kaizencc — CHANGES_REQUESTED** [review]
Hi @humanzz, this all looks pretty reasonable and you've got a nice looking README + unit tests. Just a few minor things and then I'm ready to ship

File: packages/@aws-cdk-testing/framework-integ/test/aws-cloudwatch/test/integ.table-widget.ts
[the remark this answers is not in the document]
Comment (humanzz) [reply packages/@aws-cdk-testing/framework-integ/test/aws-cloudwatch/test/integ.table-widget.ts:1]: I've only generated the snapshots, but haven't really looked in console.
let me do that as part of addressing the other comments

File: packages/aws-cdk-lib/aws-cloudwatch/lib/graph.ts [hunk 1]
```diff
@@ -519,6 +519,306 @@ export class GraphWidget extends ConcreteWidget {
...
+  AVERAGE = 'AVG',
+}
+
+/**
+ * Properties for TableWidget's summary columns
+ */
+export interface TableSummaryProps {
+  /**
+   * Summary columns
+   * @see TableSummary
```
[the remark this answers is not in the document]
Comment (humanzz) [reply packages/aws-cdk-lib/aws-cloudwatch/lib/graph.ts]: I removed it

File: packages/@aws-cdk-testing/framework-integ/test/aws-cloudwatch/test/integ.table-widget.ts
[the remark this answers is not in the document]
Comment (humanzz) [reply packages/@aws-cdk-testing/framework-integ/test/aws-cloudwatch/test/integ.table-widget.ts:1]: Here are a couple of screenshots for the dashboard/metrics deployed

Screenshot 2024-02-13 at 19 19 10
Screenshot 2024-02-13 at 19 19 40
Screenshot 2024-02-13 at 19 20 15

[the remark this answers is not in the document]
Comment (kaizencc) [reply packages/@aws-cdk-testing/framework-integ/test/aws-cloudwatch/test/integ.table-widget.ts:1]: I assume that is what you expected!

**Review by kaizencc — APPROVED** [review]
Nice! Thanks @humanzz. The API looks good to me, happy to send this one off

**Pull request merged**